tap_zendesk/http.py

```diff
@@ -208,7 +210,80 @@ def get_offset_based(url, access_token, request_timeout, page_size, **kwargs):
         yield response_json
         next_url = response_json.get('next_page')
 
-def get_incremental_export(url, access_token, request_timeout, start_time):
+@backoff.on_exception(backoff.expo,
+                    (ConnectionError, ConnectionResetError, Timeout, ChunkedEncodingError, ProtocolError),#As ConnectionError error and timeout error does not have attribute status_code,
+                    max_tries=5, # here we added another backoff expression.
+                    factor=2)
+async def call_api_async(session, url, request_timeout, params, headers):
+    """
+    Perform an asynchronous GET request
+    """
+    while True:
```

Comment:
Using `while True:` loop might lead into infinite loop if for some reason we don't receive `200` status code. I think we should have some max. retry limits applied here. Also can't we raise custom exceptions and handle these retries in backoff logic itself?

Reply:
Removed while loop and added backoff mechanism to retry 5 times.